lhotse/features/ssl.py

```diff
@@ -0,0 +1,86 @@
+from dataclasses import asdict, dataclass
+from typing import Any, Dict, Optional
+
+import numpy as np
+import torch
+
+from lhotse.features.base import FeatureExtractor, register_extractor
+from lhotse.utils import (
+    EPSILON,
+    LOG_EPSILON,
+    Seconds,
+    compute_num_frames,
+    is_module_available,
+)
+
+
+@dataclass
+class S3PRLSSLConfig:
+    sampling_rate: int = 16000
+    ssl_model: str = "wav2vec2_large_ll60k"
+    layer: int = -1
+    device: str = "cpu"
+
+    def to_dict(self) -> Dict[str, Any]:
+        return asdict(self)
+
+    @staticmethod
+    def from_dict(data: Dict[str, Any]) -> "S3PRLSSLConfig":
+        return S3PRLSSLConfig(**data)
+
+
+@register_extractor
+class S3PRLSSL(FeatureExtractor):
+    name = "s3prl-ssl"
+    config_type = S3PRLSSLConfig
+
+    def __init__(self, config: Optional[Any] = None):
+        super().__init__(config)
+        assert is_module_available(
+            "s3prl.hub"
+        ), "To use s3prl ssl extractors, please install s3prl first."
+        import s3prl.hub as hub
+
+        assert self.config.ssl_model in dir(
+            hub
+        ), f"S3PRL dose not suport model: {self.config.ssl_model}."
+
+        ssl_model = getattr(hub, self.config.ssl_model)()
+        self.ssl_model = ssl_model.to(self.config.device)
+
+    def feature_dim(self, sampling_rate: int) -> int:
+        assert (
+            sampling_rate == 16000
+        ), f"All the upstream models in S3PRL now only support 16 kHz audio."
+        feature_dim = 1024
```

Comment:
You might want to move feature_dim, sampling_rate, and frame_shift to the config to let people override it for different models (even if today they all use the same setting)